### ISSUE TITLE
LogRotatorSink not handling correctly the last element (before completion)

### ISSUE THREAD

**Issue by adrian-salajan** [akka/alpakka#2553]
```
  val done =
    Source(1 to 4)
      .map(i => ByteString.fromString(i.toString))

      .toMat(
        LogRotatorSink.withSinkFactory(
          triggerGeneratorCreator =
            () => n => Some(Path.of(new String(s"file${n.decodeString("UTF-8")}" + ".txt"))),
          sinkFactory =
            (path: Path) =>
              Flow[ByteString].toMat(FileIO.toPath(path, Set(CREATE, WRITE, TRUNCATE_EXISTING, SYNC)))(Keep.right)
        )
      )(Keep.right).run

  Await.ready(done, 2.seconds)
  system.terminate()
  Await.ready(system.whenTerminated, 2.seconds)
```

Expected result:
file1.txt, contents: 1 
file2.txt, contents: 2 
file3.txt, contents: 3
file4.txt, contents: 4

Actual result:
file1.txt, contents: 1 
file2.txt, contents: 2 
file3.txt, contents: 3
file4.txt, contents: 

Last file is always created but empty.

version "akka-stream-alpakka-file" % "2.0.2"

**Comment by tg44** [comment]
This is fixed by the PR above.

**Comment by ennru** [comment]
This was fixed with #2559 in 3.0.0-M1.
